### PR TITLE
Added the ability to install Win32-OpenSSH from Github

### LINKED ISSUE #621: OpenSSH install fails after windows defender definition failures
**What steps did you take and what happened:**

Since we have allowed the windows updates to fail I am periodically seeing the ssh install fail immediately after,  I believe these may be related but unsure what is causing this at the moment, it is almost like the command leaves the host in an errored state meaning installing features fails.  
```    
    vsphere: fatal: [default]: FAILED! => {"changed": true, "cmd": "$service = Get-Service \"Windefend\"\n$service.WaitForStatus(\"Running\",\"00:5:00\")\nUpdate-MpSignature", "delta": "0:00:29.765605", "end": "2021-05-28 07:48:05.548545", "msg": "non-zero return code", "rc": 1, "start": "2021-05-28 07:47:35.782940", "stderr": "Update-MpSignature : Virus and spyware definitions update was completed with errors.\r\nAt line:3 char:1\r\n+ Update-MpSignature\r\n+ ~~~~~~~~~~~~~~~~~~\r\n    + CategoryInfo          : NotSpecified: (MSFT_MpSignature:ROOT\\Microsoft\\...SFT_MpSignature) [Update-MpSignature], \r\n    CimException\r\n    + FullyQualifiedErrorId : HRESULT 0x80072efe,Update-MpSignature", "stderr_lines": ["Update-MpSignature : Virus and spyware definitions update was completed with errors.", "At line:3 char:1", "+ Update-MpSignature", "+ ~~~~~~~~~~~~~~~~~~", "    + CategoryInfo          : NotSpecified: (MSFT_MpSignature:ROOT\\Microsoft\\...SFT_MpSignature) [Update-MpSignature], ", "    CimException", "    + FullyQualifiedErrorId : HRESULT 0x80072efe,Update-MpSignature"], "stdout": "", "stdout_lines": []}
```

    vsphere: ...ignoring

    vsphere:

    vsphere: TASK [systemprep : Install OpenSSH] ********************************************

    vsphere: fatal: [default]: FAILED! => {"changed": true, "cmd": "Add-WindowsCapability -Online -Name OpenSSH.Server~~~~0.0.1.0", "delta": "0:00:12.453111", "end": "2021-05-28 07:48:35.439165", "msg": "non-zero return code", "rc": 1, "start": "2021-05-28 07:48:22.986053", "stderr": "Add-WindowsCapability : Add-WindowsCapability failed. Error code = 0x80072efe\r\nAt line:1 char:65\r\n+ ... ing $false; Add-WindowsCapability -Online -Name OpenSSH.Server~~~~0.0 ...\r\n+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\r\n    + CategoryInfo          : NotSpecified: (:) [Add-WindowsCapability], COMException\r\n    + FullyQualifiedErrorId : Microsoft.Dism.Commands.AddWindowsCapabilityCommand", "stderr_lines": ["Add-WindowsCapability : Add-WindowsCapability failed. Error code = 0x80072efe", "At line:1 char:65", "+ ... ing $false; Add-WindowsCapability -Online -Name OpenSSH.Server~~~~0.0 ...", "+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~", "    + CategoryInfo          : NotSpecified: (:) [Add-WindowsCapability], COMException", "    + FullyQualifiedErrorId : Microsoft.Dism.Commands.AddWindowsCapabilityCommand"], "stdout": "", "stdout_lines": []}

I am wondering if a reboot after the update steps will get the node always in a state to receive the openssh package?   Also @jsturtevant is there a location we can download the openssh cab to avoid this having to be an online install every time?  this would help in situations where we are building nodes in an internet restricted environment

**Environment:**

Project ([Image Builder for Cluster API](https://github.com/kubernetes-sigs/image-builder/tree/master/images/capi), [kube-deploy/imagebuilder](https://github.com/kubernetes-sigs/image-builder/tree/master/images/kube-deploy/imagebuilder), [konfigadm](https://github.com/kubernetes-sigs/image-builder/tree/master/images/konfigadm)):

Additional info for Image Builder for Cluster API related issues:

- OS (e.g. from `/etc/os-release`, or `cmd /c ver`): windows 2019
- Packer Version: 1.7
- Packer Provider: vsphere
- Ansible Version: 
- Cluster-api version (if using):
- Kubernetes version: (use `kubectl version`):

/kind bug
/sig windows

### PR DESCRIPTION
**Pull request by perithompson**
What this PR does / why we need it:
Installing OpenSSH from a Windows feature will not work when building in air-gapped environments.  To achieve this, this PR adds the ability to specify a ssh_source_url for the OpenSource Win32-OpenSSH port available on Github.

Which issue(s) this PR fixes: 
Fixes #621 

**Additional context**
If no ssh_source_url is specified OpenSSH will be installed from MSFT updates as normal so no changes are required in CI. This also has the ability to avoid the Error code = 0x80072f8f failure when Windows Defender updates errors during the playbook run

/sig windows

### CODE REVIEW

**Comment by hxietkg** [comment]
Hope this PR merged soon

**Comment by perithompson** [comment]
@jsturtevant are you able to take a look at this please?

**Comment by jsturtevant** [comment]
Sorry for the delay was on a long vacation!

/lgtm 
/approve

**Review by codenrhoden — CHANGES_REQUESTED** [review]
I just have the one comment on the copyright year change that I don't think is necessary.

Going through this, I'm struck that our process could use improvement because I think @jsturtevant's `/approve` should have been enough. We want Windows to be able to iterate independently and more quickly. I'm thinking we should add an `OWNERS` file to allow the `image-builder-windows-maintainers` alias approval rights in both `packer/config/windows` and `docs/`. Right now, both of these are defaulting to just the top-level `OWNERS` file, which doesn't seem like what we want here. Thoughts?

File: images/capi/ansible/windows/roles/systemprep/tasks/main.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2020 The Kubernetes Authors.
+# Copyright 2021 The Kubernetes Authors.
```
[the remark this answers is not in the document]
Comment (jsturtevant) [reply images/capi/ansible/windows/roles/systemprep/tasks/main.yml:1]: I am not sure on the guidance, I have been asked to update it in the past.

**Comment by jsturtevant** [comment]
> Going through this, I'm struck that our process could use improvement because I think @jsturtevant's `/approve` should have been enough.

agree, in addition I've been blocked on the prs if it requires updates to the `makefile`. I don't know the best solution but adding the permissions you suggested would be a good start.

File: images/capi/ansible/windows/roles/systemprep/tasks/main.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (perithompson) [reply images/capi/ansible/windows/roles/systemprep/tasks/main.yml:1]: I've been asked a couple of times in the k8s codebase to update the copyright when I've modified the file.  I wonder if there is anything telling us when to update this copyright notice?

[the remark this answers is not in the document]
Comment (nikhita) [reply images/capi/ansible/windows/roles/systemprep/tasks/main.yml:1]: > As a general rule, we don't update the copyright year when files are modified.

This is correct. :+1: 

Copyright year is not updated when files are modified.

[the remark this answers is not in the document]
Comment (perithompson) [reply images/capi/ansible/windows/roles/systemprep/tasks/main.yml]: removed this copyright change, thanks @nikhita

**Comment by perithompson** [comment]
@jsturtevant @codenrhoden PTAL

**Pull request merged**